### PR TITLE
ZK-4595: support a way to re-sort a ListModel under MVVM pattern

### PR DESCRIPTION
**Pull request by DevChu**

### CODE REVIEW

**Review by DevChu — COMMENTED** [review]
extract to  interface Sortable<T> 
default void sort() {};

**Review by scribetw — CHANGES_REQUESTED** [review]
sort

**Review by scribetw — COMMENTED** [review]
Add tests

File: zktest/test/java/org/zkoss/zktest/zats/test2/F96_ZK_4595Test.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * @author jameschu
+ */
+public class F96_ZK_4595Test extends WebDriverTestCase {
+	@Test
+	public void test() {
+		connect();
+		click(jq("$header1"));
+		click(jq("$header2"));
+		click(jq("$header3"));
+		click(jq("$header5"));
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/F96_ZK_4595Test.java:35]: add header4 but comment out

**Review by scribetw — CHANGES_REQUESTED** [review]

**Review by scribetw — APPROVED** [review]

**Pull request merged**